hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatReader.java

```diff
@@ -141,4 +139,8 @@ public HoodieLogBlock prev() throws IOException {
     return this.currentReader.prev();
   }
 
+  private Schema getReaderSchema() {
+    boolean useWriterSchema = !readerSchema.isEmptySchema();
+    return useWriterSchema ? null : readerSchema.getAvroSchema();
```

Comment:
This is so confusing, why we use writer schema if the reader schema is not empty ? And what does it means for an empty internal schema ?

Reply:
I used variable here on purpose to point out `useWriterSchema`
Essentially, it represents removed code
https://github.com/apache/hudi/blob/release-0.12.1/hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java#L162-L169

Reply:
Then please also keep the old comments.

Reply:
You are right, the old comment should have been kept. I fixed that in `add nullable annotation and comment about writer schema`